src/components/ColorSwatches/index.js

```diff
@@ -0,0 +1,144 @@
+import styled from 'styled-components';
+import Button from '../../components/Button';
+
+function ColorSwatches({
+  currentColor,
+  setDesktopState,
+  setOpenColorSwatches,
+}) {
+  const [customColor, setCustomColor] = useState(false);
+  const [highlighted, setHighlighted] = useState(currentColor);
+
+  const sampleColumn1 = ['#ffffff', '#fa0000', '#00fc00', '#0602fd', '#bddbc3'];
+  const sampleColumn2 = ['#000000', '#7c0001', '#027f00', '#040078', '#a3c9ee'];
+  const sampleColumn3 = ['#bfbfc1', '#fcfb09', '#00fcfd', '#fd06f5', '#fcf9f0'];
+  const sampleColumn4 = ['#818181', '#7f7e09', '#03807a', '#641c5c', '#a2a1a7'];
+
+  const colorsArray = [
+    sampleColumn1,
+    sampleColumn2,
+    sampleColumn3,
+    sampleColumn4,
+  ];
+
+  useEffect(() => {
+    const isSystemColor = checkIfSystemColor(currentColor, colorsArray);
+    if (isSystemColor) {
+      setCustomColor(false);
+    } else {
+      setCustomColor(currentColor);
+    }
+  }, [currentColor]);
+
+  const checkIfSystemColor = (colorToCheck, systemColors) => {
+    let existSystemColor = false;
+    for (const colorCode of systemColors) {
+      if (colorCode.includes(colorToCheck)) existSystemColor = true;
+    }
+    return existSystemColor;
+  };
+
+  const handleColorSelect = selectedColor => {
+    setDesktopState(prev => ({ ...prev, color: selectedColor }));
+    setOpenColorSwatches(false);
+  };
+
+  return (
+    <Swatches>
+      <section className="inline-grid">
+        {colorsArray.map(column => (
+          <div key={column[0]}>
+            {column.map(color => (
+              <SampleBox
+                onClick={() => handleColorSelect(color)}
+                key={color}
+                sampleColor={color}
+                highlighted={color === highlighted}
+                onMouseEnter={() => setHighlighted(color)}
+              />
+            ))}
+          </div>
+        ))}
+      </section>
+      <hr />
+      <section className="inline-grid">
+        <Button>
+          <input
+            type="color"
+            className="customColorPicker"
+            onChange={e => handleColorSelect(e.target.value)}
+          />
+          Other...
+        </Button>
+        <SampleBox
+          hidden={!customColor}
+          sampleColor={customColor}
+          highlighted={customColor === highlighted}
+          onMouseEnter={() => setHighlighted(customColor)}
+          onClick={() => handleColorSelect(customColor)}
+        />
+      </section>
+    </Swatches>
+  );
+}
+
+const Swatches = styled.div`
+  position: absolute;
+  width: 95px;
+  height: 153px;
+  padding-left: 1px;
+  background-color: rgb(236, 233, 218);
+  box-sizing: border-box;
+  outline: 1px outset white;
+  -webkit-box-shadow: 2px 2px 3px -1px rgba(0, 0, 0, 0.75);
+  -moz-box-shadow: 2px 2px 3px -1px rgba(0, 0, 0, 0.75);
+  box-shadow: 2px 2px 3px -1px rgba(0, 0, 0, 0.75);
+
+  & .customColorPicker {
+    position: absolute;
+    width: 61px !important;
```

Comment:
`!important`, here and below in line 116, is almost always a "code smell". I see this was added due to another rule with higher specificity; Please look up how to increase specificity with `styled-components` and utilize another solution 🙏